### PR TITLE
[WIP] Support command line parameters to add an additional layer of configurability in the documentation/code generator.

### PR DESCRIPTION
**Pull request by lquerel**
Closing: #193

### CODE REVIEW

File: crates/weaver_forge/src/filter.rs [hunk 1]
```diff
@@ -17,8 +16,9 @@ pub struct Filter {
 impl Filter {
     /// Create a new filter from a string expression or return an error if the
     /// expression is invalid.
-    pub fn try_new(filter_expr: &str) -> Result<Self, Error> {
-        let vars = Vec::new();
+    /// The vars parameter is a list of variable names that can be used in the
+    /// filter expression.
+    pub fn try_new(filter_expr: &str, vars: Vec<String>) -> Result<Self, Error> {
```
Comment (jsuereth) [inline crates/weaver_forge/src/filter.rs:21]: Nit: Should we pass in a `Map<String,Value>` here for vars?  It feels wierd to have two lists that are dependent on each other in different places.

File: crates/weaver_forge/src/lib.rs [hunk 2]
```diff
@@ -95,6 +98,37 @@ impl Object for TemplateObject {
...
+        for (key, value) in params {
+            _ = new_params.insert(key, Value::from_serialize(&value));
+        }
+        Self { params: new_params }
+    }
+}
+
+impl Display for ParamsObject {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.write_str(&format!("{:#?}", self.params))
```
Comment (jsuereth) [inline crates/weaver_forge/src/lib.rs:120]: Nit:  I'm not a fan of relying on debug formatting...  Can/should we create a formatter for HashMap and use it here and/or just a helper method?

File: crates/weaver_forge/src/filter.rs (hunk 1 above)
Comment (lquerel) [reply crates/weaver_forge/src/filter.rs:21]: I agree, it's weird but it's the jaq API...

File: crates/weaver_forge/src/lib.rs (hunk 2 above)
Comment (lquerel) [reply crates/weaver_forge/src/lib.rs:120]: This `Display` implementation is unused. I will remove it in the next commit.

**Review by jsuereth — APPROVED** [review]

**Pull request merged**